src/main/java/com/senkiv/bookstore/validation/FieldMatchValidator.java

```diff
@@ -0,0 +1,71 @@
+package com.senkiv.bookstore.validation;
+
+import com.senkiv.bookstore.exception.FieldValidationException;
+import com.senkiv.bookstore.validation.annotation.FieldMatch;
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Objects;
+
+public class FieldMatchValidator implements ConstraintValidator<FieldMatch, Object> {
+    public static final String COULD_NOT_ACCESS_FIELD_ON_CLASS =
+            "Could not access field '%s' on class '%s'";
+    private String firstFieldName;
+    private String secondFieldName;
+
+    @Override
+    public void initialize(FieldMatch constraintAnnotation) {
+        this.firstFieldName = constraintAnnotation.first();
+        this.secondFieldName = constraintAnnotation.second();
+    }
+
+    @Override
+    public boolean isValid(Object value, ConstraintValidatorContext context) {
+        if (value == null) {
```

Comment:
you can simplify this method (example)
```suggestion
        Object field = new BeanWrapperImpl(value).getPropertyValue(this.field);
        Object fieldMatch = new BeanWrapperImpl(value).getPropertyValue(this.fieldMatch);
        return Objects.equals(field, fieldMatch);
```